frontend/src/metabase/admin/people/components/UserGroupSelect.jsx

```diff
@@ -1,58 +1,70 @@
-/* eslint "react/prop-types": "warn" */
 import React, { Component } from "react";
 import PropTypes from "prop-types";
+import Group from "metabase/entities/groups";
+import { connect } from "react-redux";
+
+import { createMembership, deleteMembership } from "../people";
+import { getGroupsWithoutMetabot, getUserMemberships } from "../selectors";
 
 import LoadingSpinner from "metabase/components/LoadingSpinner";
 
 import GroupSelect from "./GroupSelect";
 
+@Group.loadList()
+@connect(
+  (state, props) => ({
+    groups: getGroupsWithoutMetabot(state, props),
+    userMemberships: getUserMemberships(state, props),
+  }),
+  {
+    createMembership,
+    deleteMembership,
+  },
+)
 export default class UserGroupSelect extends Component {
   static propTypes = {
-    user: PropTypes.object.isRequired,
+    userId: PropTypes.string.isRequired,
+    isCurrentUser: PropTypes.bool.isRequired,
+    userMemberships: PropTypes.array,
     groups: PropTypes.array,
     createMembership: PropTypes.func.isRequired,
     deleteMembership: PropTypes.func.isRequired,
-    isCurrentUser: PropTypes.bool.isRequired,
-  };
-
-  static defaultProps = {
-    isInitiallyOpen: false,
   };
 
-  toggle() {
-    this.refs.popover.toggle();
-  }
-
-  render() {
+  handleGroupChange = (group, isAdded) => {
     const {
-      user,
-      groups,
+      userId,
+      userMemberships,
       createMembership,
       deleteMembership,
-      isCurrentUser,
     } = this.props;
 
-    if (!groups || groups.length === 0 || !user.memberships) {
+    if (isAdded) {
+      createMembership({ groupId: group.id, userId });
+    } else {
+      const membershipId = userMemberships.find(m => m.group_id === group.id)
+        .membership_id;
+
+      deleteMembership({ membershipId });
+    }
+  };
+
+  render() {
+    const { groups, userMemberships, isCurrentUser } = this.props;
+
+    if (!groups || groups.length === 0 || !userMemberships) {
       return <LoadingSpinner />;
     }
```

Comment:
Is it possible there are no groups and `groups.length === 0` is part of a _loaded_ behavior?
Entity loaders provide loading props like `loading` and `loaded`, should we maybe rely them?